gulp/typescript/typescript-definition-generator.js

```diff
@@ -0,0 +1,317 @@
+/**
+ * This script generates a type definition by taking JSDoc roughly parsed data,
+ * formatting it and passing it to a mustache template.
+ */
+
+const fs = require('fs');
+const mustache = require('mustache');
+
+// Retrieve JSDoc data.
+const data = JSON.parse(fs.readFileSync(__dirname + '/typescript-definition-data.json', 'utf8'));
+const classes = data.classes;
+let globals = data.global.properties;
+
+// Format classes.
+classes.forEach(cls => {
+    // Format class.
+    // Store name as `className` and not simply `name`, to avoid name conflict
+    // in static constructors block.
+    cls.className = cls._name;
+    // Store closest parent if there is one.
+    cls.extends = cls.inheritsFrom && cls.inheritsFrom.length > 0
+        ? cls.inheritsFrom[0]
+        : null;
+    // Store comment using class tag as description.
+    cls.comment = formatComment(cls.comment, 'class');
+
+    // Build a filter for deprecated or inherited methods or properties.
+    const filter = _ => !_.deprecated && _.memberOf == cls.alias && !_.isNamespace;
```

Comment:
@sasensi I think this use of `_` for these method arguments is rather confusing. I first thought you're using the underscore library :) 

Could you replace it with `it`? I've seen this being used elsewhere, it's short, and it's verbally correct, as you're iterating over an array and each entry is `it`. Sounds good?

Reply:
Ok, I pushed the change.

Reply:
Thanks! I just released `v0.12.0`, kind of a shame I didn't include this in it as well... Will  merge now!